### PR TITLE
New Features for TFLite Delegates accuracy and correctness tools

### PR DESCRIPTION
**Pull request by robert-kalmar**
The new features are: 

- **Option to use reference and test model for the evaluation tools.** This supports the cases, where the delegate does not perform inline model compilation for target compute IP but requires model to be preprocessed in advance.  For instance the workflow with Ethos-U NPU - the model is converted by Arm Vela tool, which replaces part of the compute graph with custom nodes recognized by the particular delegate. To evaluate the accuracy in this case one needs to supply both the reference and converted (test) model. The reference to be used with reference inference and test model with delegate. 
-  **Add support for int32 input/output**

### CODE REVIEW

**Review by qukhan — APPROVED** [review]

File: tensorflow/lite/tools/evaluation/stages/inference_profiler_stage.h [hunk 1]
```diff
@@ -32,13 +32,15 @@ namespace evaluation {
...
-// 1. User-defined TfliteInferenceParams (The 'test' setting)
-// 2. Default TfliteInferenceParams (The 'reference' setting)
-// The latter essentially implies single-threaded CPU execution.
+// 1. The 'test' setting: User-defined TfliteInferenceParams
+// 2. The 'reference' setting: User-defined or default TfliteInferenceParams
+// The default essentially implies single-threaded CPU execution.
 class InferenceProfilerStage : public EvaluationStage {
  public:
   explicit InferenceProfilerStage(const EvaluationStageConfig& config)
       : EvaluationStage(config) {}
```
Comment (qukhan) [inline tensorflow/lite/tools/evaluation/stages/inference_profiler_stage.h:41]: When this constructor is called, the tool fails when trying to initialize the `reference_stage_`.

The error is:
```
E0618 02:09:57.517596    3384 tflite_inference_stage.cc:109] TfliteInferenceParams not provided
tensorflow/lite/tools/evaluation/stages/inference_profiler_stage_test.cc:67: Failure
```

Maybe you need to initialize the reference config as it is done [here][1]?

[1]: https://github.com/tensorflow/tensorflow/pull/62937/files#diff-8e9daef4424254eca5ebca48cddcb99b8c2b2ae16a80dd817d396edfe1b02a57L83-L91

**Review by qukhan — CHANGES_REQUESTED** [review]

**Comment by keerthanakadiri** [comment]
Hi @qukhan , Any update on this PR ? Please. Thank you!

**Comment by keerthanakadiri** [comment]
Hi @@qukhan , Any update on this PR ? Please. Thank you!

**Comment by keerthanakadiri** [comment]
Hi @robert-kalmar,  Can you please look into the [comment](https://github.com/tensorflow/tensorflow/pull/62937#discussion_r1646184784) from @qukhan

**Pull request closed without merging**